### ISSUE TITLE
[SOLUTION] Hardware fix on CSI to mitigate HDMI backpowering for AIMOS KVM's

### ISSUE THREAD

**Issue by arch1mede** [pikvm/pikvm#382]
This requires ADVANCED soldering to perform on CSI module

![image](https://user-images.githubusercontent.com/3489071/124841978-65e15080-df43-11eb-848a-26427e63ec3e.png)
![image](https://user-images.githubusercontent.com/3489071/124841987-6a0d6e00-df43-11eb-822e-fdde6b21c7ed.png)

Required: Diode 1N4148 (cathode of the diode faces away from pin 15)
Notes: You need to interrupt the v3.3 input and put a diode on it.  So have to break the circuit, then bridge the break with the diode. if the traces are buried that can be hard to do

OR

![image](https://user-images.githubusercontent.com/3489071/124841852-13a02f80-df43-11eb-80af-45a7c130d327.png)

Replace the 0-Ohm with the above diode

**Comment by woffko** [comment]
I have different KVM but the same problem — when KVM is plugged into CSI HDMI board, Pi4 refuses to boot.  Unfortunately this solution didn't work, but I have my own. Replace the other fuse, near the HDMI socket with the same 1N4148, black side of the diode towards the HDMI connector. 
Before
![backp](https://user-images.githubusercontent.com/2505149/128005405-a05ca0ad-542a-4d06-8fff-ecded4e2817b.jpg)
After
![backp2](https://user-images.githubusercontent.com/2505149/128006923-046e3eb6-7763-4fd3-9fe1-9fa77192ca49.jpg)

**Comment by mdevaev** [comment]
@woffko which KVM do you use? Model?

**Comment by woffko** [comment]
Currently I've tried Inter-Tech KVM AS-21HA IPC HDMI, Aimos aka https://www.aliexpress.com/item/4001230679168.html?spm=a2g0s.9042311.0.0.7f094c4djuveFW and Ugreen https://www.aliexpress.com/item/32854516883.html?spm=a2g0s.9042311.0.0.27424c4dAbvWcX
I have two PCs with asrock rack boards, one is J1900D2Y and the second is E3C226D2I
J1900D2Y is very problematic, because it should be connected to the monitor or directly to the pikvm upon cold boot or you just don't have any output on HDMI, don't know what the problem is, but the second PC with E3C226D2I board works without any problem via VGA-HDMI converter.
In conclusion  Inter-Tech KVM AS-21HA IPC HDMI works, but you need to switch it manually, hotkeys are not working.
Ugreen works the same way as Inter-tech, only manual switch between sources.
Aimos that I've ordered by clicking on recommended noname Ali HDMI switch works, but I have no luck —  the PIKVM keyboard and virtual media are not working at all, tried all ports, but no luck.

**Comment by ronaldlw** [comment]
> 
> 
> Currently I've tried [Inter-Tech KVM AS-21HA IPC HDMI](https://www.inter-tech.de/en/products/ipc/kvm/kvm-as-21ha), [Aimos](https://www.aliexpress.com/item/4001230679168.html?spm=a2g0s.9042311.0.0.7f094c4djuveFW) and [Ugreen](https://www.aliexpress.com/item/32854516883.html?spm=a2g0s.9042311.0.0.27424c4dAbvWcX)
> I have two PCs with asrock rack boards, one is J1900D2Y and the second is E3C226D2I
> J1900D2Y is very problematic, because it should be connected to the monitor or directly to the pikvm upon cold boot or you just don't have any output on HDMI, don't know what the problem is, but the second PC with E3C226D2I board works without any problem via VGA-HDMI converter connected to HDMI switch.
> In conclusion Inter-Tech KVM AS-21HA IPC HDMI works, but you need to switch it manually, hotkeys are not working.
> Ugreen works the same way as Inter-tech, only manual switch between sources.
> Aimos that I've ordered by clicking on recommended noname Ali HDMI switch works, but I have no luck — the PIKVM keyboard and virtual media are not working at all, tried all ports, but no luck.

I'm confused.  This sounds like what you're saying is that this "fix" doesn't work at all with any of the kvms you've tried?  I wouldn't consider "works, but requires manual switch between sources" to in any way be working, am I misunderstanding?

**Comment by woffko** [comment]
> 
> 
> > Currently I've tried [Inter-Tech KVM AS-21HA IPC HDMI](https://www.inter-tech.de/en/products/ipc/kvm/kvm-as-21ha), [Aimos](https://www.aliexpress.com/item/4001230679168.html?spm=a2g0s.9042311.0.0.7f094c4djuveFW) and [Ugreen](https://www.aliexpress.com/item/32854516883.html?spm=a2g0s.9042311.0.0.27424c4dAbvWcX)
> > I have two PCs with asrock rack boards, one is J1900D2Y and the second is E3C226D2I
> > J1900D2Y is very problematic, because it should be connected to the monitor or directly to the pikvm upon cold boot or you just don't have any output on HDMI, don't know what the problem is, but the second PC with E3C226D2I board works without any problem via VGA-HDMI converter connected to HDMI switch.
> > In conclusion Inter-Tech KVM AS-21HA IPC HDMI works, but you need to switch it manually, hotkeys are not working.
> > Ugreen works the same way as Inter-tech, only manual switch between sources.
> > Aimos that I've ordered by clicking on recommended noname Ali HDMI switch works, but I have no luck — the PIKVM keyboard and virtual media are not working at all, tried all ports, but no luck.
> 
> I'm confused. This sounds like what you're saying is that this "fix" doesn't work at all with any of the kvms you've tried? I wouldn't consider "works, but requires manual switch between sources" to in any way be working, am I misunderstanding?

**Backpowering  issue is fixed with all listed KVMs**, but I decided to share my experience regarding all the functions of the KVMs, not just backpowering issue. Sorry for the misunderstanding.

**Comment by gotshadowbantempaccount** [comment]
Actually got same problem with Aimos and HDMI
1. when power is on on "USB 5V cable to mouse port" and then plug power to Raspberry PI original power supply 5.1V its refuse to boot. But wteh power on raspberyPI then USB power tpo HDMI its working
2. when use some cheap power supply raspbery pi only boot when is HDMi CSI 2 is disconnected
3. For me solution for now is to "power on first by remote reboot socket" 1) raspbery PI power then 2)+5V USB to mouse port
4. **Another solution will be maybe try to block power on HDMi cable same like on USB so AMIOS will be power on only from USB cable connected to keyboard port**

![CCC](https://user-images.githubusercontent.com/88304290/134068121-c07084f6-d954-49ba-8286-822da69a026e.png)

![DDD2](https://user-images.githubusercontent.com/88304290/134067996-3c833c97-932f-46cd-b551-a06f405a934b.jpg)

**Comment by arch1mede** [comment]
> 
> 
> Actually got same problem with Aimos and HDMI
> 
>     1. when power is on on "USB 5V cable to mouse port" and then plug power to Raspberry PI original power supply 5.1V its refuse to boot. But wteh power on raspberyPI then USB power tpo HDMI its working
> 
>     2. when use some cheap power supply raspbery pi only boot when is HDMi CSI 2 is disconnected
> 
>     3. For me solution for now is to "power on first by remote reboot socket" 1) raspbery PI power then 2)+5V USB to mouse port
> 
>     4. **Another solution will be maybe try to block power on HDMi cable same like on USB so AMIOS will be power on only from USB cable connected to keyboard port because maybe is try to get power from raspberry pi and refuse to boot (unconvifrmed)**
> 
> 
> ![CCC](https://user-images.githubusercontent.com/88304290/134068121-c07084f6-d954-49ba-8286-822da69a026e.png)
> 
> ![DDD2](https://user-images.githubusercontent.com/88304290/134067996-3c833c97-932f-46cd-b551-a06f405a934b.jpg)

Your actually the first to report that. We have a bunch of active users who have AIMOS KVM's and don't have a mouse port issue like you are porting.

**Comment by ronaldlw** [comment]
1) you shouldn't be applying power via the mouse port.  is that what you are saying you're doing here??
> 
> 
> Actually got same problem with Aimos and HDMI
> 
>     1. when power is on on "USB 5V cable to mouse port" and then plug power to Raspberry PI original power supply 5.1V its refuse to boot. But wteh power on raspberyPI then USB power tpo HDMI its working
> 
>     2. when use some cheap power supply raspbery pi only boot when is HDMi CSI 2 is disconnected
> 
>     3. For me solution for now is to "power on first by remote reboot socket" 1) raspbery PI power then 2)+5V USB to mouse port
> 
>     4. **Another solution will be maybe try to block power on HDMi cable same like on USB so AMIOS will be power on only from USB cable connected to keyboard port because maybe is try to get power from raspberry pi and refuse to boot (unconvifrmed)**
> 
> 
> ![CCC](https://user-images.githubusercontent.com/88304290/134068121-c07084f6-d954-49ba-8286-822da69a026e.png)
> 
> ![DDD2](https://user-images.githubusercontent.com/88304290/134067996-3c833c97-932f-46cd-b551-a06f405a934b.jpg)

**Comment by arch1mede** [comment]
If there is power being delivered on the mouse port, that's back power and this needs to be taken care of. Short term you will only experience odd behaviors, long term you are looking at equipment damage to the usb port or rpi or target or all 3.

**Comment by gotshadowbantempaccount** [comment]
@arch1mede   @ronaldlw  problem is only with "power on" **raspebery pi can't power on** (boot)  when AIMOS in **on** before raspberry. try to disable HDMI power Pin but that not help. 

So for me now only solution is to:
1) power on first raspbery pi by "remote socket"
2) power on second socket that give power to "AIMOS" by USB

**Comment by arch1mede** [comment]
> 
> 
> @arch1mede @ronaldlw problem is only with "power on" **raspebery pi can't power on** (boot) when AIMOS in **on** before raspberry. try to disable HDMI power Pin but that not help.
> 
> So for me now only solution is to:
> 
>     1. power on first raspbery pi by "remote socket"
> 
>     2. power on second socket that give power to "AIMOS" by USB

This sounds suspiciously like you have a wiring issue. No one else has reported this and this def sounds like back power. Please join discord and provide pics of your setup.
